model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/entities/RoleByNameQuery.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2025 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.cache.infinispan.entities;
+
+import org.keycloak.models.RealmModel;
+
+import java.util.Set;
+
+/**
+ * @author Alexander Schwartz
+ * @version $Revision: 1 $
+ */
+public class RoleByNameQuery extends AbstractRevisioned implements RoleQuery, InClient {
+    private final String role;
+    private final String realm;
+    private final String realmName;
```

Comment:
`realmName` does not seem to be used. Same for `RoleListQuery`.

Reply:
It is implementing the `InRealm` interface which needs it. This is used to remove entries that belong to a realm, see for example 

https://github.com/keycloak/keycloak/blob/18a6c790113998bf039f46d6c71ffb668e961036/model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/stream/InRealmPredicate.java#L41

Reply:
Isn't that the realm ID?

Reply:
Ok, I stand corrected. I'll create a follow-up PR on this one once @pedroigor approves the rest of the PR.